### DISCUSSION TITLE
Inconsistency on latency numbers reported by Prometheus and X-Kong-Proxy-Latency

### DISCUSSION THREAD

**Discussion by Sanil210898** [Kong/kong#10803] [Help] [answered]
I have been trying to debug latency related issues. Numbers I am seeing on prometheus for kong_latency_ms_bucket vary with the numbers sent in the response header 'X-Kong-Proxy-Latency'. The numbers in 'X-Kong-Proxy-Latency' seem more accurate to me, whereas Prometheus numbers seem very high.

I am unsure which number to trust and what do these mean. From the docs, both should be the same. Can someone help me in understanding the differences?

**Comment by samugi** [comment, marked as answer]
Hi @Sanil210898 , 
the explanation should be that while `X-Kong-Proxy-Latency` represents the time taken by Kong to process a request **before** proxying the request upstream, [the kong_latency_ms_bucket](https://github.com/Kong/kong/blob/0d3561b984dfee8bfe9037d86d3ba1e52cdda1c6/kong/plugins/prometheus/exporter.lua#L353) also [includes the response time](https://github.com/Kong/kong/blob/0d3561b984dfee8bfe9037d86d3ba1e52cdda1c6/kong/pdk/log.lua#L828-L829) ((response +) header_filter + body_filter).
